### PR TITLE
spec: Fix listener issue in RTL7c

### PR DESCRIPTION
**Pull request by mattheworiordan**
Fixes https://github.com/ably/docs/issues/226

@paddybyers @tcard @SimonWoolf you happy for me to proceed?

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Note I did not change the IDL to reflect the optional callback as I don't think it's a necessary part of the IDL and arguably is the result of `=> io`.  Shout if you disagree

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -467,7 +467,7 @@ h3(#realtime-channel). Channel
 * @(RTL7)@ @Channel#subscribe@ function:
 ** @(RTL7a)@ Subscribe with no arguments subscribes a listener to all messages
 ** @(RTL7b)@ Subscribe with a single name argument subscribes a listener to only messages whose @name@ member matches the string name
-** @(RTL7c)@ Implicitly attaches the @Channel@ if the channel is in the @INITIALIZED@ state. However, if the channel is in or enters the @DETACHED@ or @FAILED@ state before the operation succeeds, it will result in the listener not being registered and an error being indicated, typically to the optional callback where the language permits
+** @(RTL7c)@ Implicitly attaches the @Channel@ if the channel is in the @INITIALIZED@ state. However, if the channel is in or enters the @DETACHED@ or @FAILED@ state before the operation succeeds, it will an error being indicated, typically to the optional callback where the language permits. The listener will always be registered regardless of the implicit attach result
```
Comment (paddybyers) [inline content/client-lib-development-guide/features.textile]: ^ what he said

Comment (mattheworiordan) [inline content/client-lib-development-guide/features.textile]: > If the attach fails, an error will be indicated, typically to an optional callback where the language permits.

Surely we should mention it calls the optional callback upon success too?

**Comment by mattheworiordan** [comment]
@paddybyers @SimonWoolf how about 86df673 instead?

**Comment by paddybyers** [comment]
LGTM

**Comment by SimonWoolf** [comment]
:+1:

**Pull request merged**